src/pudl/output/ferc1.py

```diff
@@ -871,3 +873,417 @@ def add_mean_cap_additions(steam_df):
+            + ["xbrl_factoid_name_original"],
+            dropna=False,
+        )[[value_col]]
+        .nunique()
+    )
+    assert inter_table_ref_check[inter_table_ref_check[value_col] > 1].empty
+
+    factoid_idx = ["xbrl_factoid_name_original", "table_name"]
+    exploded = exploded.set_index(factoid_idx)
+    inter_table_refs = exploded.loc[
+        inter_table_facts_to_drop.set_index(factoid_idx).index
+    ]
+    if len(inter_table_refs) > 1:
+        logger.info(
+            f"Explode: Dropping {len(inter_table_refs)} ({len(inter_table_refs)/len(exploded):.01%}) inter-table references."
+        )
+        exploded = exploded.loc[
+            exploded.index.difference(
+                inter_table_facts_to_drop.set_index(factoid_idx).index
+            )
+        ].reset_index()
+    else:
+        logger.info("Explode: Found no inter-table references. Dropping nothing.")
+        # reset the index bc our index rn is tbl name and factoid name og.
+        exploded = exploded.reset_index()
+    return exploded
+
+
+def get_table_level(table_name: str, top_table: str) -> int:
+    """Get a table level."""
+    # we may be able to infer this nesting from the metadata
+    table_nesting = {
+        "balance_sheet_assets_ferc1": {
+            "utility_plant_summary_ferc1": {
+                "plant_in_service_ferc1": None,
+                "electric_plant_depreciation_changes_ferc1": None,
+            },
+        },
+        "balance_sheet_liabilities_ferc1": {"retained_earnings_ferc1": None},
+        "income_statement_ferc1": {
+            "depreciation_amortization_summary_ferc1": None,
+            "electric_operating_expenses_ferc1": None,
+            "electric_operating_revenues_ferc1": None,
+        },
+    }
+    if table_name == top_table:
+        level = 1
+    elif table_name in table_nesting[top_table].keys():
+        level = 2
+    elif table_name in pudl.helpers.dedupe_n_flatten_list_of_lists(
+        [values.keys() for values in table_nesting[top_table].values()]
+    ):
+        level = 3
+    else:
+        raise AssertionError(
+            f"AH we didn't find yer table name {table_name} in the nested group of "
+            "tables. Be sure all the tables you are trying to explode are related."
+        )
+    return level
+
+
+def get_table_levels(tables_to_explode: list[str], top_table: str) -> pd.DataFrame:
+    """Get a set of table's level in the explosion.
+
+    Level in this context means where it sits in the tree of the relationship of these
+    tables. Level 1 is at the root while the higher numbers are towards the leaves of
+    the trees.
+    """
+    table_levels = {"table_name": [], "table_level": []}
+    for table_name in tables_to_explode:
+        table_levels["table_name"].append(table_name)
+        table_levels["table_level"].append(get_table_level(table_name, top_table))
+    return pd.DataFrame(table_levels)
+
+
+def remove_totals_from_other_dimensions(
```

Comment:
Other dimensions here refers to the sub-total columns.
Currently this method drops "totals" for any table that actively has sub-dimensions.
We currently aren't correcting for sub-total v. total differences, which we should probably do if we're going to drop these.